build/dockerfiles/rhel.Dockerfile

```diff
@@ -0,0 +1,118 @@
+    if [[ -f /tmp/root-local.tgz ]] || [[ ${BOOTSTRAP} == "true" ]]; then \
+      mkdir -p /root/.local; tar xf /tmp/root-local.tgz -C /root/.local/; rm -fr /tmp/root-local.tgz;  \
+      /usr/bin/pip3.6 install --user yq jsonschema; \
+      # could be installed in /opt/app-root/src/.local/bin or /root/.local/bin
+      for d in /opt/app-root/src/.local /root/.local; do \
+        if [[ -d ${d} ]]; then \
+          cp ${d}/bin/yq ${d}/bin/jsonschema /usr/local/bin/; \
+          pushd ${d}/lib/python3.6/site-packages/ >/dev/null; \
+            cp -r PyYAML* xmltodict* yaml* yq* jsonschema* /usr/lib/python3.6/site-packages/; \
+          popd >/dev/null; \
+        fi; \
+      done; \
+      chmod -c +x /usr/local/bin/*; \
+    else \
+      /usr/bin/pip3.6 install yq jsonschema; \
+    fi && \
+    ln -s /usr/bin/python3.6 /usr/bin/python && \
+    # test install worked
+    for d in python yq jq jsonschema; do echo -n "$d: "; $d --version; done
+
+# for debugging only
+# RUN microdnf install -y util-linux && whereis python pip jq yq && python --version && jq --version && yq --version
+
+# Registry, organization, and tag to use for base images in dockerfiles. Devfiles
+# will be rewritten during build to use these values for base images.
+ARG PATCHED_IMAGES_REG="quay.io"
+ARG PATCHED_IMAGES_ORG="eclipse"
+ARG PATCHED_IMAGES_TAG="nightly"
+
+COPY /build/scripts /arbitrary-users-patch/base_images /build/
+COPY /devfiles /build/devfiles
+WORKDIR /build/
+RUN TAG=${PATCHED_IMAGES_TAG} \
+    ORGANIZATION=${PATCHED_IMAGES_ORG} \
+    REGISTRY=${PATCHED_IMAGES_REG} \
+    ./update_devfile_patched_image_tags.sh
+RUN ./check_mandatory_fields.sh devfiles
+RUN ./index.sh > /build/devfiles/index.json
+
+################# 
+# PHASE TWO: configure registry image
+################# 
+
+# Build registry, copying meta.yamls and index.json from builder
+# UPSTREAM: use RHEL7/RHSCL/httpd image so we're not required to authenticate with registry.redhat.io
+# https://access.redhat.com/containers/?tab=tags#/registry.access.redhat.com/rhscl/httpd-24-rhel7
+FROM registry.access.redhat.com/rhscl/httpd-24-rhel7:2.4-104 AS registry
+
+# DOWNSTREAM: use RHEL8/httpd
+# https://access.redhat.com/containers/?tab=tags#/registry.access.redhat.com/rhel8/httpd-24
+# FROM registry.redhat.io/rhel8/httpd-24:1-60 AS registry
+USER 0
+
+# BEGIN these steps might not be required
+RUN sed -i /etc/httpd/conf/httpd.conf \
+    -e "s,Listen 80,Listen 8080," \
+    -e "s,logs/error_log,/dev/stderr," \
+    -e "s,logs/access_log,/dev/stdout," \
+    -e "s,AllowOverride None,AllowOverride All," && \
+    chmod a+rwX /etc/httpd/conf /run/httpd
+STOPSIGNAL SIGWINCH
+# END these steps might not be required
+
+RUN mkdir /var/www/html/devfiles
+COPY .htaccess README.md /var/www/html/
+COPY --from=builder /build/devfiles /var/www/html/devfiles
+RUN chmod -R g+rwX /var/www/html/devfiles
+COPY /build/scripts/*entrypoint.sh /usr/local/bin/
```

Comment:
See `build/dockerfiles/entrypoint.sh` and `build/dockerfiles/rhel.entrypoint.sh` (merged in #110). I think `rhel.entrypoint.sh` covers the functionality of `uid_entrypoint` and `entrypoint` from this PR.